opendatahub-io/odh-manifests
# Fix dashboard auth for service accounts

## Description
Previously, when using the `authentication` overlay for the dashboard, authn/z didn't work for OpenShift service accounts. I discovered that this was due to a misconfigured `--openshift-delegate-urls` value being passed to the dashboard. This change updates the argument to a valid value so that service accounts can be used with the dashboard.

## How Has This Been Tested?
This has been tested on an openshift 4.11 cluster by performing the following:

* Deploy the ODH operator to your cluster from the operator catalog
* Create a namespace for testing this change:
```
oc new-project auth-fix
```
* Create a kfdef to deploy the current version:
```
cat <<EOF | oc apply -f -
apiVersion: kfdef.apps.kubeflow.org/v1
kind: KfDef
metadata:
  name: opendatahub
  namespace: auth-fix
spec:
  applications:
    - kustomizeConfig:
        overlays:
          - authentication
        repoRef:
          name: manifests
          path: odh-dashboard
      name: odh-dashboard
  repos:
    - name: manifests
      uri: 'https://github.com/opendatahub-io/odh-manifests/tarball/master'
EOF
```
* Run the following command to attempt to log into the dashboard using the `odh-dashboard` service account. The command should output `403`, indicating that the service account has been restricted from accessing the dashboard:
```
curl --header "Authorization: Bearer $(oc get secret -n auth-fix -o jsonpath='{.data.token}' `oc get secret -n auth-fix | grep odh-dashboard-token | awk '{print $1}'` | base64 -d)" -s -o /dev/null -w "%{http_code}\n" -k https://$(oc get route -n auth-fix odh-dashboard -o jsonpath='{.spec.host}')
```
* Update the kfdef to deploy this change by running the following:
```
cat <<EOF | oc apply -f -
apiVersion: kfdef.apps.kubeflow.org/v1
kind: KfDef
metadata:
  name: opendatahub
  namespace: auth-fix
spec:
  applications:
    - kustomizeConfig:
        overlays:
          - authentication
        repoRef:
          name: manifests
          path: odh-dashboard
      name: odh-dashboard
  repos:
    - name: manifests
      uri: 'https://github.com/accorvin/idh-manifests/tarball/fix-dashboard-oauth'
EOF
```
* Run the curl command again. This time it should print a status code of `200`, indicating that the service account has successfully reached the dashboard:
```
curl --header "Authorization: Bearer $(oc get secret -n auth-fix -o jsonpath='{.data.token}' `oc get secret -n auth-fix | grep odh-dashboard-token | awk '{print $1}'` | base64 -d)" -s -o /dev/null -w "%{http_code}\n" -k https://$(oc get route -n auth-fix odh-dashboard -o jsonpath='{.spec.host}')
```
* Navigate to the dashboard in a browser. Confirm that you can log in and use the dashboard as normal.

## Merge criteria:
